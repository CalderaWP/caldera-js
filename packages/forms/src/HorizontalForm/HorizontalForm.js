import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { CalderaForm } from '../CalderaForm/CalderaForm';

/**
 * Displays a form, one field per row
 *
 * @param fields
 * @param initialValues
 * @param onClose
 * @param onChange
 * @param instanceId
 * @return {*}
 * @constructor
 */
export const HorizontalForm = ({
	fields,
	initialValues,
	onClose,
	onChange,
	instanceId,
								   conditionals
}) => {
	const rows = [];
	let i = 0;
	fields.forEach(field => {
		rows.push({
			rowId: `${instanceId}-${i}`,
			columns: [
				{
					columnId: `c-${instanceId}-${i}`,
					fields: [field],
					width: '1'
				}
			]
		});
		i++;
	});
	const form = {
		id: `horizontal-form-${instanceId}`,
		fields,
		rows,
		conditionals
	};

	return (
		<Fragment>
			<CalderaForm
				form={form}
				initialValues={initialValues}
				onSubmit={onClose}
				onChange={onChange}
			/>
		</Fragment>
	);
};

HorizontalForm.propTypes = {
	fields: PropTypes.array,
	initialValues: PropTypes.object,
	onSubmit: PropTypes.func,
	onChange: PropTypes.func,
	onBlur: PropTypes.func,
	onClose: PropTypes.func,
	initialValues: PropTypes.object,
	conditionals: PropTypes.array,
};

HorizontalForm.defaultProps = {
	fields: [],
	initialValues: {},
	conditionals: [],
};
