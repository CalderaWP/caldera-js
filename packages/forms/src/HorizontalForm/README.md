# Horizontal Form
This form takes an array of field and creates a grid that had one field in one full-width column per row.
