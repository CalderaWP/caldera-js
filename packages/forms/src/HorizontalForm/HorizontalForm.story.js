import React, {Component} from 'react';
import { storiesOf } from '@storybook/react';
import { HorizontalForm } from './HorizontalForm';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField,
	radioField
} from '@calderawp/components';


storiesOf('HorizontalForm', module).add('Works with a bunch of fields', () => (
	<HorizontalForm
		fields={[
			checkboxFieldset,
			selectField,
			checkboxField,
			numberField,
			textField,
			emailField,
			radioField
		]}
		onChange={values => console.log(values)}
		onBlur={values => console.log(values)}
		onClose={values => console.log(values)}
	/>
));

