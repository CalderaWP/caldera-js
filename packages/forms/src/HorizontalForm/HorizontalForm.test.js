import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { HorizontalForm } from './HorizontalForm';

import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField,
	radioField
} from '@calderawp/components';

describe('HorizontalForm', () => {
	let onClose;
	let onChange;
	let onBlur;

	beforeEach(() => {
		onClose = jest.fn();
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('Matches snapshot', () => {
		const component = renderer.create(
			<HorizontalForm
				fields={[checkboxFieldset, selectField, checkboxField]}
				initialValues={{}}
				onClose={onClose}
				onChange={onChange}
				onBlur={onBlur}
				instanceId={'test-1'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Calls on change', () => {
		const component = mount(
			<HorizontalForm
				fields={[checkboxFieldset, selectField, checkboxField]}
				initialValues={{}}
				onClose={onClose}
				onChange={onChange}
				onBlur={onBlur}
				instanceId={'test-2'}
			/>
		);
		const event = { target: { value: 2 } };
		component
			.find('select')
			.first()
			.simulate('change', event);
		expect(onChange.mock.calls.length).toBe(1);
	});
});
