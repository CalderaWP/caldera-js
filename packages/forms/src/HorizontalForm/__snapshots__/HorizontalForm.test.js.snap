// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalForm Matches snapshot 1`] = `
<div>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="sc-bdVaJa caldera-row caldera-form-row sc-bwzfXH pjOvZ"
      id={undefined}
    >
      <div
        className="caldera-column sc-bdVaJa iHZvIS"
        id={undefined}
        width={undefined}
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <fieldset
            className="caldera-fieldset caldera-fieldset-checkboxes"
          >
            <legend>
              Checkbox Field Set Label
            </legend>
            <div
              className="components-base-control"
            >
              <div
                className="components-base-control__field"
              >
                <input
                  aria-describedby={undefined}
                  checked={true}
                  className="components-checkbox-control__input"
                  id="opt-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value="1"
                />
                <label
                  className="components-checkbox-control__label"
                  htmlFor="inspector-checkbox-control-0"
                >
                  One
                </label>
              </div>
            </div>
            <div
              className="components-base-control"
            >
              <div
                className="components-base-control__field"
              >
                <input
                  aria-describedby="inspector-checkbox-control-1__help"
                  checked={true}
                  className="components-checkbox-control__input"
                  id="opt-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value="1"
                />
                <label
                  className="components-checkbox-control__label"
                  htmlFor="inspector-checkbox-control-1"
                >
                  Two
                </label>
              </div>
              <p
                className="components-base-control__help"
                id="inspector-checkbox-control-1__help"
              >
                The Second Option
              </p>
            </div>
            <div
              className="components-base-control"
            >
              <div
                className="components-base-control__field"
              >
                <input
                  aria-describedby={undefined}
                  checked={true}
                  className="components-checkbox-control__input"
                  id="opt-3"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value="1"
                />
                <label
                  className="components-checkbox-control__label"
                  htmlFor="inspector-checkbox-control-2"
                >
                  Three
                </label>
              </div>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa caldera-row caldera-form-row sc-bwzfXH pjOvZ"
      id={undefined}
    >
      <div
        className="caldera-column sc-bdVaJa iHZvIS"
        id={undefined}
        width={undefined}
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control caldera-field caldera-field-select"
          >
            <div
              className="components-base-control__field"
            >
              <label
                className="components-base-control__label"
                htmlFor="inspector-select-control-0"
              >
                Select Field Label
              </label>
              <select
                aria-describedby="inspector-select-control-0__help"
                className="components-select-control__input"
                id="selectFieldId"
                multiple={false}
                onChange={[Function]}
                value={undefined}
              >
                <option
                  value={0}
                >
                  Zero
                </option>
                <option
                  value={1}
                >
                  One
                </option>
                <option
                  value={2}
                >
                  Two
                </option>
              </select>
            </div>
            <p
              className="components-base-control__help"
              id="inspector-select-control-0__help"
            >
              Select field description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa caldera-row caldera-form-row sc-bwzfXH pjOvZ"
      id={undefined}
    >
      <div
        className="caldera-column sc-bdVaJa iHZvIS"
        id={undefined}
        width={undefined}
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control"
          >
            <div
              className="components-base-control__field"
            >
              <input
                aria-describedby="inspector-checkbox-control-3__help"
                checked={true}
                className="components-checkbox-control__input"
                id="checkboxFieldLabel"
                onBlur={[Function]}
                onChange={[Function]}
                type="checkbox"
                value="1"
              />
              <label
                className="components-checkbox-control__label"
                htmlFor="inspector-checkbox-control-3"
              >
                Checkbox Labe
              </label>
            </div>
            <p
              className="components-base-control__help"
              id="inspector-checkbox-control-3__help"
            >
              Checkbox field description
            </p>
          </div>
        </div>
      </div>
    </div>
    <input
      className="caldera-forms-submit"
      disabled={false}
      type="submit"
      value="Click Button"
    />
  </form>
</div>
`;
